utilix/rundb.py

```diff
@@ -558,11 +560,21 @@ def _collection(experiment, collection, url=None, user=None, password=None, data
     if not database:
         database = uconfig.get('RunDB', f'{experiment}_database')
 
-    uri = f"mongodb://{user}:{password}@{url}"
-    c = pymongo.MongoClient(uri, readPreference='secondaryPreferred')
-    DB = c[database]
-    coll = DB[collection]
-    return coll
+    # build other client kwargs
+    max_pool_size = uconfig.get('RunDB', 'max_pool_size', fallback=100)
+    socket_timeout = uconfig.get('RunDB', 'socket_timeout', fallback=60000)
+    connect_timeout = uconfig.get('RunDB', 'connect_timeout', fallback=60000)
+
+    global MONGO_CLIENTS
+    if experiment not in MONGO_CLIENTS:
+        uri = f"mongodb://{user}:{password}@{url}"
+        MONGO_CLIENTS[experiment] = pymongo.MongoClient(uri, readPreference='secondaryPreferred',
```

Comment:
Storing the clients under experiment name is incompatible with the function signature. Either store a client for any uri or change to signature to disallow passing your own username/pass. I would suggest:
```python
uri = f"mongodb://{user}:{password}@{url}"
if (experiment,uri) not in MONGO_CLIENTS:
    MONGO_CLIENTS[(experiment,uri)] = pymongo.MongoClient(...
db = MONGO_CLIENTS[(experiment,uri)][database]

```